internal/widget/scroller.go

```diff
@@ -479,26 +574,26 @@ func computeOffset(start, delta, outerWidth, innerWidth float32) float32 {
 	return offset
 }
 
+// AddScrollOffsetChangedListener allows Fyne internal code to get an inside notification when a scroller moves.
+// TODO find a way to make this generally available
+func AddScrollOffsetChangedListener(s *ScrollContainer, f func()) {
```

Comment:
Could this be a method in ScrollContainer?
```suggestion
func (s *ScrollContainer) SetOffsetChanged(callback func()) {
```
Or maybe onOffsetChanged could be exported? (exports from `internal/` are only accessible inside the project right?)

Reply:
That is why I left the TODO in there.
I wanted to get the refactor done with the understanding we knew we were special.
I would like to come back to it later, so that the refactoring PRs are not adding new public APIs.